wfe/web-front-end.go

```diff
@@ -998,11 +1008,19 @@ func (wfe *WebFrontEndImpl) Certificate(response http.ResponseWriter, request *h
 			if strings.HasPrefix(err.Error(), "gorp: multiple rows returned") {
 				wfe.sendError(response, "Multiple certificates with same short serial", err, http.StatusConflict)
 			} else {
-				wfe.sendError(response, "Not found", err, http.StatusNotFound)
+				response.Header().Add("Cache-Control", "public, max-age=0, no-cache")
+				wfe.sendError(response, "Certificate not found", err, http.StatusNotFound)
 			}
 			return
 		}
 
+		// Set cache-control header if certificate NotAfter is > time.Now().Add(-WFE.CertCacheExpirationWindow)
+		if time.Now().Add(-wfe.CertNoCacheExpirationWindow).After(cert.Expires) {
```

Comment:
Actually, looking closer I realized this is for the `/cert/<serial>` URLs, whose content is never supposed to change (even after the cert expires). So let's just make this a static max-age of one year or so, regardless of cert expiration time.